package.json

```diff
@@ -51,9 +51,35 @@
   },
   "lint-staged": {
```

Comment:
@westonruter Since there are multiple entries for `lint-staged`, should we switch to `.lintstagedrc.json`? 

Also, if we add `.lintstagedrc.json` in each plugin, then it will auto-resolve the staged files in that particular directory. For the experiment I put `plugins/webp-uploads/.lintstagedrc.json` and when I made changes to the files in `plugins/webp-uploads` it automatically ran on staged files within `plugins/webp-uploads`.

// path: `plugins/webp-uploads/.lintstagedrc.json`
```json
{
  "*.php": [
    "composer run-script lint --working-dir=../../",
    "composer run-script phpstan --working-dir=../../"
  ],
  "*.js": [
    "npm run lint-js"
  ]
}
```

output:
```sh
➜  performance git:(trunk) ✗ git commit -m "Temp commit"   
✔ Preparing lint-staged...
✔ Preparing lint-staged...
❯ Running tasks for staged files...
  ❯ plugins/webp-uploads/.lintstagedrc.json — 1 file
    ❯ *.php — 1 file
      ✔ composer run-script lint --working-dir=../../
      ✖ composer run-script phpstan --working-dir=../../
    ↓ *.js — no files [SKIPPED]
◼ Applying modifications from tasks...
◼ Cleaning up temporary files...
``

Reply:
Hummm, this doesn't seem to work. I tried putting that `.lintstagedrc.json` at `plugins/webp-uploads`. I then added an intentional PHPCS failure, like putting the wrong text domain in a translation function. It didn't block the commit. I think this is because `plugins` is still excluded in the root `phpcs.xml.dist`. No PHPCS errors were reported. The PHPStan checks are running successfully, however.

Reply:
To get it to work, I have to explicitly run `lint:webp-uploads` and not just `lint`:

```json
{
  "*.php": [
    "composer run-script lint:webp-uploads --working-dir=../../",
    "composer run-script phpstan --working-dir=../../"
  ],
  "*.js": [
    "npm run lint-js"
  ]
}
```

However, I'm not sure this is better than having a single root config because only the lint command is plugin-specific currently. Having plugin-specific `.lintstagedrc.json` files would introduce more duplication.

Reply:
> However, I'm not sure this is better than having a single root config because only the lint command is plugin-specific currently. Having plugin-specific .lintstagedrc.json files would introduce more duplication.

Humm and it's a long-known trade-off in mono repos. It will bring duplication but lead to simpler management on plugin level but we can have a root level `.lintstagedrc.json` for now.

Reply:
Did some experiments on https://github.com/WordPress/performance/pull/1090 with dynamic lint-staged config. 
@westonruter Can you please test it?